framework/src/modules/chain/synchronizer/utils.js

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright © 2019 Lisk Foundation
+ *
+ * See the LICENSE file at the top-level directory of this distribution
+ * for licensing information.
+ *
+ * Unless otherwise agreed in a custom licensing agreement with the Lisk Foundation,
+ * no part of this software, including this file, may be copied, modified,
+ * propagated, or distributed except according to the terms contained in the
+ * LICENSE file.
+ *
+ * Removal or modification of this copyright notice is prohibited.
+ */
+
+'use strict';
+
+/**
+ * Restore blocks from temp table and re-apply to chain
+ * Steps:
+ * 1. Read all blocks from temp_block table
+ * 2. Apply blocks one by one to current chain
+ * 3. Each block gets deleted from temp_block table when its being applied
+ *
+ * @param {Object} tx - database transaction
+ * @return {Promise<Boolean>|Error} - returns true when successfully restoring blocks
+ */
+const restoreBlocks = async (blocksModule, processorModule, tx) => {
+	const tempBlocks = await blocksModule.getTempBlocks(tx);
+
+	if (tempBlocks.length === 0) {
+		throw new Error('Temp_block table is empty');
```

Comment:
do we need to throw error in the case it doesn't exist?
Maybe we can just return false and end the execution?